compiler/rustc_span/src/lib.rs

```diff
@@ -1331,6 +1331,86 @@ impl SourceFileHash {
     }
 }
 
+#[derive(Copy, PartialEq, PartialOrd, Clone, Ord, Eq, Hash, Debug, Encodable, Decodable)]
+pub enum DebuggerVisualizerType {
+    Natvis,
+}
+
+rustc_data_structures::impl_stable_hash_via_hash!(DebuggerVisualizerType);
+
+/// A single debugger visualizer file.
+#[derive(Clone, PartialEq, Eq)]
+pub struct DebuggerVisualizerFile {
+    /// The name of the debugger visualizer file.
+    pub name: FileName,
+    /// The complete debugger visualizer source.
+    pub src: Option<Lrc<String>>,
+    /// The debugger visualizer's hash.
+    pub src_hash: SourceFileHash,
+    /// Indicates which type
+    pub visualizer_type: DebuggerVisualizerType,
+    /// Indicates which crate this `DebuggerVisualizer` was imported from.
+    pub cnum: CrateNum,
+}
+
+rustc_data_structures::impl_stable_hash_via_hash!(DebuggerVisualizerFile);
+
+impl Hash for DebuggerVisualizerFile {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        state.write(&self.src_hash.hash_bytes());
```

Comment:
It a `Hash` impl necessary? Also this doesn't include every part of the `DebuggerVisualizerFile` in the hash.

Reply:
Now using `#[derive(Hash)]` instead